src/log/Log.cc

```diff
@@ -416,16 +395,20 @@ void Log::_flush(EntryQueue *t, EntryQueue *requeue, bool crash)
       }
 
       if (do_fd) {
-        line[line_used] = '\n';
-        if (need_dynamic) {
-          _write_and_copy(line, line_used + 1);
-        }
+	line[line_used] = '\n';
+	if (need_dynamic) {
```

Comment:
[OT] nit, if this is in our critical path, does it make sense to put something like
```c++
if (unlikely(need_dynamic)) {
  // ...
}
```